config/obstacles.json

```diff
@@ -3,1233 +3,1206 @@
+		"offsetGraphicsInX" : 261,
+		"offsetGraphicsInY" : 129,
 		"blockedTiles" :  [27, 28, 43, 44, 60, 61, 76, 77, 93, 94, 109, 110, 126, 127, 142, 143, 159],
-		"defname" : "ObBhL14.pcx"
+		"graphics" : "ObBhL14.pcx",
+		"canBeRemoved" : false,
 	},
 	{
-		"id" : 28,
-		"allowedTerrain" : [],
-		"specialBattlefields" : [0],
-		"width" : 180,
-		"height" : 154,
+		"type" : "static",
+		"position" : 0,
+		"battlefieldSurface" : [1],
+		"offsetGraphicsInX" : 180,
+		"offsetGraphicsInY" : 154,
 		"blockedTiles" :  [22, 38, 39, 40, 44, 45, 46, 55, 56, 57, 62, 63, 123, 124, 125, 130, 131, 140, 141, 146, 147, 148],
-		"defname" : "ObBhL16.pcx"
+		"graphics" : "ObBhL16.pcx",
+		"canBeRemoved" : false,
 	},
 	{
-		"id" : 29,
-		"allowedTerrain" : [],
-		"specialBattlefields" : [5],
-		"width" : 304,
-		"height" : 264,
+		"type" : "static",
+		"position" : 0,
+		"battlefieldSurface" : [19],
+		"offsetGraphicsInX" : 304,
+		"offsetGraphicsInY" : 264,
 		"blockedTiles" :  [76, 77, 92, 93, 94, 95, 109, 110, 111],
-		"defname" : "ObCFL00.pcx"
+		"graphics" : "ObCFL00.pcx",
+		"canBeRemoved" : false,
 	},
 	{
-		"id" : 30,
-		"allowedTerrain" : [],
-		"specialBattlefields" : [6],
-		"width" : 256,
-		"height" : 257,
+		"type" : "static",
+		"position" : 0,
+		"battlefieldSurface" : [17],
+		"offsetGraphicsInX" : 256,
+		"offsetGraphicsInY" : 257,
 		"blockedTiles" :  [76, 77, 78, 92, 93, 94, 107, 108, 109],
-		"defname" : "ObLPL00.pcx"
+		"graphics" : "ObLPL00.pcx",
+		"canBeRemoved" : false,
 	},
 	{
-		"id" : 31,
-		"allowedTerrain" : [],
-		"specialBattlefields" : [7],
-		"width" : 257,
-		"height" : 255,
+		"type" : "static",
+		"position" : 0,
+		"battlefieldSurface" : [14],
+		"offsetGraphicsInX" : 257,
+		"offsetGraphicsInY" : 255,
 		"blockedTiles" :  [76, 77, 91, 92, 93, 94, 95, 108, 109, 110, 111],
-		"defname" : "ObFFL00.pcx"
+		"graphics" : "ObFFL00.pcx",
+		"canBeRemoved" : false,
 	},
 	{
-		"id" : 32,
-		"allowedTerrain" : [],
-		"specialBattlefields" : [8],
-		"width" : 277,
-		"height" : 218,
+		"type" : "static",
+		"position" : 0,
+		"battlefieldSurface" : [15],
+		"offsetGraphicsInX" : 277,
+		"offsetGraphicsInY" : 218,
 		"blockedTiles" :  [60, 61, 75, 76, 77, 91, 92, 93, 94, 95],
-		"defname" : "ObRLL00.pcx"
+		"graphics" : "ObRLL00.pcx",
+		"canBeRemoved" : false,
 	},
 	{
-		"id" : 33,
-		"allowedTerrain" : [],
-		"specialBattlefields" : [9],
-		"width" : 300,
-		"height" : 214,
+		"type" : "static",
+		"position" : 0,
+		"battlefieldSurface" : [16],
+		"offsetGraphicsInX" : 300,
+		"offsetGraphicsInY" : 214,
 		"blockedTiles" :  [59, 60, 74, 75, 76, 93, 94, 95, 111, 112],
-		"defname" : "ObMCL00.pcx"
+		"graphics" : "ObMCL00.pcx",
+		"canBeRemoved" : false,
+	}
+],
+"InherentObstacles" : [
+	{
+		"type" : "moat",
+		"blockedTiles" : [11, 28, 44, 61, 77, 111, 129, 146, 164, 181],
+		"damage" : 70,
+		"offsetGraphicsInX" : 410,
+		"offsetGraphicsInY" : 90,
+		"graphics" : "SgCsMoat.pcx",
+		"place" : ["Castle"]

```

Comment:
Where should modded castles moat configuration should be?

Reply:
what you mean? By default moat should be in InherentObstacles since it shouldn't be random I think.

Reply:
Each new town added in mod has own moat graphics

Reply:
You`ve removed  moat graphics handling from `getSiegeName`, but recreated only original towns configuration in obstacles.json. What to do with modded towns?

Reply:
Hmm, I will try with Cove but should work if you write ""place" : ["Cove"]" I suppose.

Reply:
It's working, you can also change the graphics if you want.
![zrzut ekranu z 2017-10-14 07-57-16](https://user-images.githubusercontent.com/13856703/31573124-6304b764-b0b5-11e7-8aa4-65fcb8ae88e4.png)

Reply:
@alexvins ok, I even extract graphics and update a hota mod for future
![zrzut ekranu z 2017-10-14 09-07-17](https://user-images.githubusercontent.com/13856703/31573591-2899559e-b0bf-11e7-94a3-496a391d4219.png)

Reply:
The problem is that you can not change `config/obstacles.json` from two mods at the same time (only one override will be actually used), each mod must have own configuration. We have moat graphics format in town options (see "config/schemas/townSiege.json").

Reply:
Old format... I deleted things that use him, since it was totally bad, it's not even about using, it's about design. I know about conflicts, there are many ways to bypass them, and back to old format is the worst. Still it's wip since there is needs to repair that conflict and random obstacle generator(now it's only generating static obstacles).

Reply:
You do not have to update all the mods yourself, you "just" need to retain backward compatibility.

Reply:
It's terrible idea for me, backward serialization is enought, I prefer update all mods, then keep that old compatibility.

Reply:
Backward compatibility for mods is even more important than save format's one.

Reply:
If all projects have such systems then nothing would develop, I can care for backwards mods compatibility but I can also update them. I can restore old code, or care and make new one, better one. Anyway I won't back to these solutions.

Reply:
Lets move discussion to forum https://forum.vcmi.eu/t/poll-about-backward-compatibility/4335